### PR TITLE
Use `nolint` directives for `undesirable_operator_linter()` lints

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Waiting for #1751

### CODE REVIEW

**Comment by AshesITR** [comment]
NB you can use `# nolint: undesirable_operator.` as well.

File: R/lint.R [hunk 1]
```diff
@@ -737,7 +733,7 @@ highlight_string <- function(message, column_number = NULL, ranges = NULL) {
   line <- fill_with(" ", maximum)
 
   lapply(ranges, function(range) {
-    substr(line, range[1L], range[2L]) <<-
+    substr(line, range[1L], range[2L]) <<- # nolint: undesirable_operator_linter.
       fill_with("~", range[2L] - range[1L] + 1L)
```
Comment (IndrajeetPatil) [inline R/lint.R:737]: Is the indentation lint here a false positive? AFAICT, the indentation should be 6 spaces here.

> Indentation should be 4 spaces but is 6 spaces.

Btw, if I try to inline this, I create `line_length_linter()` lint.

Comment (IndrajeetPatil) [reply R/lint.R:737]: cc @AshesITR

Comment (AshesITR) [reply R/lint.R]: It is, seems like the comment throws off the XPath for indentation changes.

Comment (IndrajeetPatil) [reply R/lint.R]: Thanks for confirming. I will wait for the related issue to be resolved. Then we can revisit this PR.

**Review by AshesITR — APPROVED** [review]

**Pull request merged**